src/system.rs

```diff
@@ -43,22 +44,67 @@ impl PendingTypes {
 
     /// Adds all the types from Context into given TypeStore and clears Context types for loading next set of types
     /// this is used after a schema named type/root type is loaded entirely into Context
-    pub fn update_type_store(&mut self, type_store: &mut TypeStore) -> IonSchemaResult<()> {
+    pub fn update_type_store(
+        &mut self,
+        type_store: &mut TypeStore,
+        load_isl_import: Option<&IslImport>,
+    ) -> IonSchemaResult<()> {
         for optional_type in &self.types_by_id {
             // return an error if any of the type in types_by_id vector is None/Unresolved
             let type_def = optional_type
                 .to_owned()
                 .ok_or(unresolvable_schema_error_raw(
                     "Unable to load schema due to unresolvable type",
                 ))?;
-            match type_def.to_owned() {
-                // Some(name) => type_store.add_named_type(name, type_def),
-                // None => type_store.add_anonymous_type(type_def),
-                TypeDefinition::Named(named_type_def) => type_store.add_named_type(named_type_def),
-                TypeDefinition::Anonymous(anonymous_type_def) => {
-                    type_store.add_anonymous_type(anonymous_type_def)
-                }
-            };
+
+            if load_isl_import.is_some() {
+                match type_def.to_owned() {
+                    TypeDefinition::Named(named_type_def) => match load_isl_import.unwrap() {
+                        IslImport::Schema(_) => type_store.add_import_type(named_type_def),
```

Comment:
I'm not sure I follow the flow here; the `_` in `Schema(_)` should be an ID that leads us to add lots of types to the `type_store`. Instead, we're adding a single named `TypeDefinition` and ignoring the ID?

I think it might help to break this into two functions.